guillotina/api/service.py

```diff
@@ -45,11 +45,40 @@ class Service(View):
     __validator__ = __schema__ = None
     _sentinal = object()
 
+    def _validate_parameters(self):
+        if 'parameters' in self.__config__:
+            data = self.request.url.query
+            for parameter in self.__config__['parameters']:
+                if parameter['in'] == 'query':
+                    if parameter['schema']['type'] == 'integer':
+                        try:
+                            int(data[parameter['name']])
+                        except ValueError:
+                            raise HTTPPreconditionFailed(content={
+                                'reason': 'Schema validation error',
+                                'message': 'can not convert {} to Int'.format(data[parameter['name']])
+                            })
+                    elif parameter['schema']['type'] == 'float':
+                        try:
+                            float(data[parameter['name']])
+                        except ValueError:
+                            raise HTTPPreconditionFailed(content={
+                                'reason': 'Schema validation error',
+                                'message': 'can not convert {} to Float'.format(data[parameter['name']])
+                            })
+                    else:
+                        pass
+                    if parameter['required'] and parameter['name'] not in data:
```

Comment:
what if they don't define `required`?

Maybe the whole check should be wrapped in `KeyError` handling?